File: TelerikUIExamples/MyAnnotation.m
//
//  MyAnnotation.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

#import "MyAnnotation.h"

@implementation MyAnnotation
{
    TKShape *_shape;
    CGPoint _center;
}

- (id)initWithShape:(TKShape*)shape X:(id)xValue Y:(id)yValue forSeries:(TKChartSeries *)series
{
    self = [super initWithX:xValue Y:yValue forSeries:series];
    if (self) {
        _shape = shape;
    }
    return self;
}

- (void)layoutInRect:(CGRect)bounds
{
    _center = [self locationInRect:bounds];
    _center.x -= _shape.size.width/2;
    _center.y -= _shape.size.height/2;
}

- (void)drawInContext:(CGContextRef)context
{
    NSMutableArray *drawables = [NSMutableArray new];
    if (_fill) {
        [drawables addObject:_fill];
    }
    if (_stroke) {
        [drawables addObject:_stroke];
    }
    [_shape drawInContext:context withCenter:_center drawings:drawables];
}


@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/MyAnnotation.swift
//
//  MyAnnotation.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class MyAnnotation:TKChartPointAnnotation {
    
    var center: CGPoint = CGPoint.zeroPoint
    var shape: TKShape?
    var fill: TKFill?
    var stroke: TKStroke?
    
    override init() {
        super.init()
    }
    
    init(Shape shape: TKShape, X xValue: AnyObject, Y yValue: AnyObject, forSeries series: TKChartSeries) {
        super.init(x: xValue, y: yValue, forSeries: series)
        self.shape = shape
    }

    override func layoutInRect(bounds: CGRect) {
        center = self.locationInRect(bounds)
        if let s = shape {
            center.x -= s.size.width/2
            center.y -= s.size.height/2
        }
    }
    
    override func drawInContext(context: CGContext!) {
        if let s = shape {
            var drawables = [TKDrawing]()
            drawables.append(fill!)
            drawables.append(stroke!)
            s.drawInContext(context, withCenter: center, drawings: drawables)
        }
    }
}